### PR TITLE
Adds information about the persistence layer of kessel-inventory

### PR DESCRIPTION
**Pull request by josejulio**
- ER diagram
 - lifecycle diagrams
 - other basic information

### CODE REVIEW

File: src/content/docs/start-here/kessel-inventory.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+It relies on services reporting the status of each resource and relations between those.
+
+To prevent services from having to save yet another id, Kessel Inventory will accept the reporter's resource id in lieu
+of any id that is stored in Kessel Inventory's database. As such, API calls that address a particular resource 
+will require the following tuple to identify a resource:
+- `local_resource_id`
+- `resource_type`
+- `reporter_id`
+- `reporter_type`
```
Comment (bvulaj) [inline src/content/docs/start-here/kessel-inventory.md:16]: Can we define these briefly?

File: src/content/docs/start-here/kessel-inventory.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    DeleteResource --> DeleteResourceExists
+    DeleteResourceExists -- No --> Fail
+    DeleteResourceExists -- Yes --> DeleteCopyResourceDo --> DeleteResourceHistory --> DeleteResourceRelationships
+```
+
+### Relationship lifecycle
+
+The lifecycle of a relationship is similar to a resource, as there is a history table where changes on the relationship are tracked.
+It follows the same approach of indicating the operation (`CREATE`, `UPDATE` and `DELETE`). 
+Note than when a resource is deleted, the relationship is also deleted, but without updating the history table for that.
```
Comment (bvulaj) [inline src/content/docs/start-here/kessel-inventory.md:190]: s/than/that

Comment (bvulaj) [inline src/content/docs/start-here/kessel-inventory.md:190]: The history table is updated at the DELETE event though, right?

File: src/content/docs/start-here/kessel-inventory.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+Lastly, we have the `local_to_inventory_id` table that maps a reporter's resource identifier to an id used in the database.
+
+### Database diagram
+
+An Entity-Relationship diagram is provided below. 
+
+Note: Entities ResourceReporter and RelationshipReporter are actually attributes in JSON format of other entities and not an entity on its own.
+
+```mermaid
```
Comment (josejulio) [inline src/content/docs/start-here/kessel-inventory.md:34]: mermaid diagrams can be seen by clicking on the 3 dots on the file and select view file

File: src/content/docs/start-here/kessel-inventory.md (hunk 2 above)
Comment (josejulio) [reply src/content/docs/start-here/kessel-inventory.md:190]: When we deleted a `resource`, the `resource_history` table gets a new `DELETED` entry. `relationships_history` table does not get any entry in this case.
When we directly delete a `relationship`, `relationship_history` tables do get a `DELETED` entry.

I'll update above to make it clear.

**Review by tonytheleg — APPROVED** [review]
This is awesome, thanks for writing this up! Super helpful!

**Pull request merged**